src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/TagHelperComponentTagHelper.cs

```diff
@@ -49,11 +50,62 @@ public abstract class TagHelperComponentTagHelper : TagHelper
             _logger = loggerFactory.CreateLogger(GetType());
         }
 
+        /// <summary>
+        /// Creates a new <see cref="TagHelperComponentTagHelper"/> and orders the 
+        /// the collection of <see cref="ITagHelperComponent"/>s in <see cref="ITagHelperComponentManager.Components"/>.
+        /// </summary>
+        /// <param name="manager">The <see cref="ITagHelperComponentManager"/> which contains the collection
+        /// of <see cref="ITagHelperComponent"/>s.</param>
+        /// <param name="propertyActivator">The <see cref="ITagHelperComponentPropertyActivator"/> activates
+        /// properties of <see cref="ITagHelperComponent"/>s.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        /// <remarks>The <see cref="ITagHelperComponentManager.Components"/> are ordered after the 
+        /// creation of the <see cref="ITagHelperComponentManager"/> to position the <see cref="ITagHelperComponent"/>s
+        /// added from controllers and views correctly.</remarks>
+        public TagHelperComponentTagHelper(
+            ITagHelperComponentManager manager,
+            ITagHelperComponentPropertyActivator propertyActivator,
+            ILoggerFactory loggerFactory)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+
+            if (propertyActivator == null)
+            {
+                throw new ArgumentNullException(nameof(propertyActivator));
+            }
+
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _components = manager.Components.OrderBy(p => p.Order).ToArray();
+            PropertyActivator = propertyActivator;
+            _logger = loggerFactory.CreateLogger(GetType());
+        }
+
+        public ITagHelperComponentPropertyActivator PropertyActivator;
+
+        [ViewContext]
+        [HtmlAttributeNotBound]
+        public ViewContext ViewContext { get; set; }
+
         /// <inheritdoc />
         public override void Init(TagHelperContext context)
         {
+            if (PropertyActivator == null)
+            {
+                var serviceProvider = ViewContext.HttpContext.RequestServices;
+                PropertyActivator = serviceProvider.GetService(typeof(ITagHelperComponentPropertyActivator))
+                    as TagHelperComponentPropertyActivator;
+            }
+
             foreach (var component in _components)
             {
+                PropertyActivator?.Activate(ViewContext, component);
```

Comment:
If `PropertyActivator` is still `null` at this point, the configuration is pretty messed up. Don't turn that messed-up-edness into a silent failure.